src/components/ProfilePage/AddHouseForm/AddHouseForm.tsx

```diff
@@ -16,90 +17,60 @@ import Typography from '@mui/material/Typography';
 import { getDownloadURL, ref, uploadBytesResumable } from 'firebase/storage';
 import { useState } from 'react';
 import { useForm } from 'react-hook-form';
-import * as yup from 'yup';
-
-import { create, storage } from '/src/firebase';
-import getRandomString from '/src/utils/getRandomString';
+import { create, storage } from 'src/firebase';
+import getRandomString from 'src/utils/getRandomString';
 
+import { addHouseFormSchema, FieldsSchema } from '../../../schemas/addHouseFormSchema';
+import FacilityCheckbox from '../FacilityCheckbox';
 import styles from './AddHouseForm.module.css';
-import FacilityCheckbox from './components/FacilityCheckbox/FacilityCheckbox';
-
-const REQUIRED_ERROR = 'This field is required.';
-const SPECIAL_CHARACTERS_ERROR = 'No special characters allowed.';
-const NUMBER_NEEDED_ERROR = 'This field must be a number.';
-const POSITIVE_NUMBER_ERROR = 'Must be a positive number.';
-const NEGATIVE_NUMBER_ERROR = 'Cannot be a negative number.';
-
-const schema = yup.object({
-  streetNumber: yup
-    .string()
-    .required(REQUIRED_ERROR)
-    .matches(/^[A-Za-z0-9 ]+$/, SPECIAL_CHARACTERS_ERROR),
-  streetName: yup.string().required(REQUIRED_ERROR),
-  streetSuffix: yup.string().required(REQUIRED_ERROR),
-  city: yup
-    .string()
-    .required(REQUIRED_ERROR)
-    .matches(/^[A-Za-z0-9 ]+$/, SPECIAL_CHARACTERS_ERROR),
-  state: yup
-    .string()
-    .required(REQUIRED_ERROR)
-    .matches(/^[A-Za-z0-9 ]+$/, SPECIAL_CHARACTERS_ERROR),
-  price: yup.number(NUMBER_NEEDED_ERROR).positive(POSITIVE_NUMBER_ERROR).required(REQUIRED_ERROR),
-  propertyType: yup.string().required(REQUIRED_ERROR),
-  yearBuilt: yup.number(NUMBER_NEEDED_ERROR).positive(POSITIVE_NUMBER_ERROR).required(REQUIRED_ERROR),
-  dimension: yup.number(NUMBER_NEEDED_ERROR).positive(POSITIVE_NUMBER_ERROR).required(REQUIRED_ERROR),
-  floor: yup.number(NUMBER_NEEDED_ERROR).min(0, NEGATIVE_NUMBER_ERROR).required(REQUIRED_ERROR),
-  floorsInBuilding: yup.number(NUMBER_NEEDED_ERROR).min(0, NEGATIVE_NUMBER_ERROR).required(REQUIRED_ERROR),
-  roomsNumber: yup.number(NUMBER_NEEDED_ERROR).positive(POSITIVE_NUMBER_ERROR).required(REQUIRED_ERROR),
-  bathroomNumber: yup.number(NUMBER_NEEDED_ERROR).positive(POSITIVE_NUMBER_ERROR).required(REQUIRED_ERROR),
-  heating: yup.string().required(REQUIRED_ERROR),
-  descriptionField: yup.string().required(REQUIRED_ERROR),
-});
 
-const AddHouseForm = () => {
+function AddHouseForm() {
   const [moreFacilitiesShown, setMoreFacilitiesShown] = useState(false);
-  const [images, setImages] = useState();
+  const [images, setImages] = useState<File[]>([]);
 
   const {
     register,
     handleSubmit,
     formState: { errors },
-  } = useForm({
+  } = useForm<FieldsSchema>({
```

Comment:
It is a good name so no need to change it but consider making the name more specific. It could be helpful to know what kind of fields schema it is just by looking at the name. It's quite easy to understand from the context and you can check where it's imported from if in doubt and making the name more specific means the name gets longer and becomes more clunky but developers can be lazy creatures so they can appreciate a longer name despite its drawbacks :wink:. I'm leaving the decision whether to change to you.